udemy_exercises/songs/Album.java

```diff
@@ -1,9 +1,217 @@
+        Scanner scanner = new Scanner(System.in);
+        boolean quit = false;
+        boolean forward = true;
+        ListIterator<Song> listIterator = playlist.listIterator();
+        if(playlist.size() == 0){
+            System.out.println("No songs in playlist.");
+            return;
+        } else{
+            System.out.println("Npw playing " + listIterator.next().toString());
+            printMenu();
+        }
+        while(!quit){
+            int action = scanner.nextInt();
+            scanner.nextLine();
+
+            switch(action){
+                case 0:
+                    stop();
+                    quit = true;
+                    break;
+                case 1:
+                    if(!forward){
+                        if(listIterator.hasNext()){
+                            listIterator.next();
+                        }
+                        forward = true;
+                    }
+                    if(listIterator.hasNext()){
+                        System.out.println("Now playing " + listIterator.next().toString());
+                    } else{
+                        System.out.println("We have reached the end of the playlist.");
+                        forward = false;
+                    }
+                    break;
+                case 2:
+                    if(forward) {
+                        if(listIterator.hasPrevious()) {
+                            listIterator.previous();
+                        }
+                        forward = false;
+                    }
+                    if(listIterator.hasPrevious()) {
+                        System.out.println("Now playing " + listIterator.previous().toString());
+                    } else {
+                        System.out.println("We are at the start of the playlist");
+                        forward = true;
+                    }
+                    break;
+                case 3:
+                    if(forward) {
+                        if(listIterator.hasPrevious()) {
+                            System.out.println("Now replaying " + listIterator.previous().toString());
+                            forward = false;
+                        } else {
+                            System.out.println("We are at the start of the list");
+                        }
+                    } else {
+                        if(listIterator.hasNext()) {
+                            System.out.println("Now replaying " + listIterator.next().toString());
+                            forward = true;
+                        } else {
+                            System.out.println("We have reached the end of the list");
+                        }
+                    }
+                    break;
+                case 4:
+                    printList(playlist);
+                    break;
+                case 5:
+                    printMenu();
+                    break;
+                case 6:
```

Comment:
After pressing "6" to delete a song, it reports the song now being played.  Then pressing "3" to replay, the song indicated is actually the _**next**_ song in the list (rather than the "current" song).  Another defect.